benchmark: No valid tuned compilation for fused_reduce with <no problem key>

@pfultz2 Seeing this error after #3097:
```
AMDMIGraphX/src/targets/gpu/compile_ops.cpp:184: benchmark: No valid tuned compilation for fused_reduce with <no problem key>
```
Repro:
(model in nas drive)
`migraphx-driver compile <nas_path>/migraphx/models/gmlp.mxr`

Full error dump:
```
In file included from /tmp/comgr-72b121/input/main.cpp:2:
In file included from /tmp/comgr-72b121/include/migraphx/kernels/index.hpp:32:
/tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:199:9: error: called object type '_Float16' is not a function or function pointer
  199 |         p2([&](auto&&... ys) {
      |         ^~
/tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:273:33: note: in instantiation of function template specialization 'migraphx::unpack_each((lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:763:13), (lambda at /tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:273:12), _Float16)::(anonymous class)::operator()<const migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>> &>' requested here
  273 |     return [=](auto f) { return f(xs...); };
      |                                 ^
/tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:198:5: note: in instantiation of function template specialization 'migraphx::pack(migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>>)::(anonymous class)::operator()<(lambda at /tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:198:8)>' requested here
  198 |     p1([&](auto&&... xs) {
      |     ^
/tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:762:9: note: in instantiation of function template specialization 'migraphx::unpack_each<(lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:763:13), (lambda at /tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:273:12), _Float16>' requested here
  762 |         unpack_each(
      |         ^
/tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:714:13: note: in instantiation of function template specialization 'migraphx::fused_reduce((lambda at /tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:273:12), migraphx::assign_none, (lambda at /tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:266:16))::(anonymous class)::operator()<migraphx::array<unsigned int, 3>, migraphx::reduce::lane::reducer<(lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:714:34)>>' requested here
  714 |             f(out_idx, make(idx, [&](auto input) { return reduce_slice<Output>(input, out_idx); }));
      |             ^
/tmp/comgr-72b121/include/migraphx/kernels/index.hpp:200:64: note: in instantiation of function template specialization 'migraphx::reduce::lane::run((lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:760:33))::(anonymous class)::operator()<unsigned int>' requested here
  200 |     static constexpr auto invoke_loop(F f, I i, D) -> decltype(f(i))
      |                                                                ^
/tmp/comgr-72b121/include/migraphx/kernels/index.hpp:244:21: note: while substituting deduced template arguments into function template 'invoke_loop' [with F = (lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:712:38), I = index_int, D = integral_constant<decltype(0), 0>]
  244 |                     invoke_loop(f, start, _c<0>);
      |                     ^
/tmp/comgr-72b121/include/migraphx/kernels/index.hpp:268:9: note: in instantiation of function template specialization 'migraphx::index::for_stride<false, (lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:712:38), migraphx::integral_constant<unsigned int, 32768>, migraphx::integral_constant<unsigned int, 32768>>' requested here
  268 |         for_stride<false>(global, n, nglobal(), f);
      |         ^
/tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:712:13: note: in instantiation of function template specialization 'migraphx::index::global_stride<(lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:712:38), migraphx::integral_constant<unsigned int, 32768>>' requested here
  712 |         idx.global_stride(nelements, [&](auto i) {
      |             ^
/tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:760:20: note: in instantiation of function template specialization 'migraphx::reduce::lane::run<migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>, (lambda at /tmp/comgr-72b121/include/migraphx/kernels/reduce.hpp:760:33)>' requested here
  760 |     Algo::template run<Reduced>([&](auto out_idx, auto r) {
      |                    ^
/tmp/comgr-72b121/input/main.cpp:25:9: note: in instantiation of function template specialization 'migraphx::fused_reduce<migraphx::reduce::lane, migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>, (lambda at /tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:273:12), migraphx::assign_none, (lambda at /tmp/comgr-72b121/include/migraphx/kernels/functional.hpp:266:16)>' requested here
   25 |         fused_reduce<reduce::lane, decltype(make_shape(index_ints<128, 1, 256>{}, index_ints<256, 256, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
      |         ^
1 error generated when compiling for gfx90a.

/workspace/pt2_benchmarks/AMDMIGraphX/src/targets/gpu/compile_hip.cpp:172: compile: hiprtc: HIPRTC_ERROR_COMPILATION: Compilation failed.
In file included from /tmp/comgr-10b917/input/main.cpp:2:
In file included from /tmp/comgr-10b917/include/migraphx/kernels/index.hpp:32:
/tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:199:9: error: called object type '_Float16' is not a function or function pointer
  199 |         p2([&](auto&&... ys) {
      |         ^~
/tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:273:33: note: in instantiation of function template specialization 'migraphx::unpack_each((lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:763:13), (lambda at /tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:273:12), _Float16)::(anonymous class)::operator()<const migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>> &>' requested here
  273 |     return [=](auto f) { return f(xs...); };
      |                                 ^
/tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:198:5: note: in instantiation of function template specialization 'migraphx::pack(migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>>)::(anonymous class)::operator()<(lambda at /tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:198:8)>' requested here
  198 |     p1([&](auto&&... xs) {
      |     ^
/tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:762:9: note: in instantiation of function template specialization 'migraphx::unpack_each<(lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:763:13), (lambda at /tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:273:12), _Float16>' requested here
  762 |         unpack_each(
      |         ^
/tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:714:13: note: in instantiation of function template specialization 'migraphx::fused_reduce((lambda at /tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:273:12), migraphx::assign_none, (lambda at /tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:266:16))::(anonymous class)::operator()<migraphx::array<unsigned int, 3>, migraphx::reduce::lane::reducer<(lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:714:34)>>' requested here
  714 |             f(out_idx, make(idx, [&](auto input) { return reduce_slice<Output>(input, out_idx); }));
      |             ^
/tmp/comgr-10b917/include/migraphx/kernels/index.hpp:200:64: note: in instantiation of function template specialization 'migraphx::reduce::lane::run((lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:760:33))::(anonymous class)::operator()<unsigned int>' requested here
  200 |     static constexpr auto invoke_loop(F f, I i, D) -> decltype(f(i))
      |                                                                ^
/tmp/comgr-10b917/include/migraphx/kernels/index.hpp:244:21: note: while substituting deduced template arguments into function template 'invoke_loop' [with F = (lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:712:38), I = index_int, D = integral_constant<decltype(0), 0>]
  244 |                     invoke_loop(f, start, _c<0>);
      |                     ^
/tmp/comgr-10b917/include/migraphx/kernels/index.hpp:268:9: note: in instantiation of function template specialization 'migraphx::index::for_stride<false, (lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:712:38), migraphx::integral_constant<unsigned int, 32768>, migraphx::integral_constant<unsigned int, 32768>>' requested here
  268 |         for_stride<false>(global, n, nglobal(), f);
      |         ^
/tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:712:13: note: in instantiation of function template specialization 'migraphx::index::global_stride<(lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:712:38), migraphx::integral_constant<unsigned int, 32768>>' requested here
  712 |         idx.global_stride(nelements, [&](auto i) {
      |             ^
/tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:760:20: note: in instantiation of function template specialization 'migraphx::reduce::lane::run<migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>, (lambda at /tmp/comgr-10b917/include/migraphx/kernels/reduce.hpp:760:33)>' requested here
  760 |     Algo::template run<Reduced>([&](auto out_idx, auto r) {
      |                    ^
/tmp/comgr-10b917/input/main.cpp:25:9: note: in instantiation of function template specialization 'migraphx::fused_reduce<migraphx::reduce::lane, migraphx::shape<migraphx::integral_const_array<unsigned int, 128, 1, 256>, migraphx::integral_const_array<unsigned int, 256, 256, 1>>, (lambda at /tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:273:12), migraphx::assign_none, (lambda at /tmp/comgr-10b917/include/migraphx/kernels/functional.hpp:266:16)>' requested here
   25 |         fused_reduce<reduce::lane, decltype(make_shape(index_ints<128, 1, 256>{}, index_ints<256, 256, 1>{}))>(y, assign_none{}, partial(MIGRAPHX_LIFT(fused_reduce_op))(xs...));
      |         ^
1 error generated when compiling for gfx90a.
```

--- a/src/fuse_reduce.cpp
+++ b/src/fuse_reduce.cpp
@@ -199,12 +199,12 @@ struct find_pointwise_reduce
 
     void apply(module_pass_manager& mpm, const match::matcher_result& r) const
     {
-        auto reduce = r.result;
+        auto reduce        = r.result;
         auto input         = r.instructions["pointwise"];
         const auto* pm     = input->module_inputs().front();
         const auto* old_rm = reduce->module_inputs().front();
 
-        auto* rm           = mpm.create_module(pm->name() + ":" + old_rm->name());
+        auto* rm = mpm.create_module(pm->name() + ":" + old_rm->name());
         rm->set_bypass();
         std::unordered_map<instruction_ref, instruction_ref> map_ins;
         // Insert pointwise
@@ -351,7 +351,7 @@ struct reduce_reshape : rewrite_reshapes_base
         auto* oldm = ins->module_inputs().front();
         auto* sm   = mpm.create_module(oldm->name() + "_reshape");
         sm->set_bypass();
-        sm->fuse(*oldm, inputs, nullptr, transform_op([&](const operation& sop) {
+        auto outs = sm->fuse(*oldm, inputs, nullptr, transform_op([&](const operation& sop) {
             if(contains(sop.name(), "reduce"))
                 return make_op(sop.name(), {{"axes", axes}});
             if(sop.name() == "multibroadcast")
@@ -359,6 +359,7 @@ struct reduce_reshape : rewrite_reshapes_base
             assert(sop.name() == "pointwise");
             return sop;
         }));
+        sm->add_return(outs);
         return mpm.get_module().insert_instruction(ins, fused_reduce{axes}, inputs, {sm});
     }
 

--- a/test/fuse_reduce.cpp
+++ b/test/fuse_reduce.cpp
@@ -895,7 +895,7 @@ TEST_CASE(reduce_reshape_reduce)
         auto y   = mm->add_parameter("y", s2);
         auto x1r = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), x1);
         auto x2r = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3r.lens()}}), x2);
-        auto yr      = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), y);
+        auto yr  = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), y);
         auto freduce = add_reduce(
             p2,
             "main:pointwise2:main:reduce_sum2_reshape_reshape:main:pointwise3_reshape:main:reduce_"
@@ -928,6 +928,108 @@ TEST_CASE(reduce_reshape_reduce)
             mm->add_instruction(migraphx::make_op("reshape", {{"dims", s2.lens()}}), freduce);
         mm->add_return({freducer});
     }
+    EXPECT(p1.sort() == p2.sort());
+}
+
+TEST_CASE(reshape_reduce_reduce_reduce_diff_axes)
+{
+    migraphx::shape s1{migraphx::shape::float_type, {128, 196, 256}};
+    migraphx::shape s2{migraphx::shape::float_type, {1}};
+    migraphx::shape s3{migraphx::shape::float_type, {25088, 256}};
+
+    migraphx::program p1;
+    {
+        auto* mm = p1.get_main_module();
+        auto x1  = mm->add_parameter("x1", s1);
+        auto x2  = mm->add_parameter("x2", s1);
+        auto l1  = mm->add_literal(migraphx::literal(s2, {1.0}));
+        auto l2  = mm->add_literal(migraphx::literal(s2, {2.0}));
+
+        auto x1_rsp = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), x1);
+        auto l2_mb =
+            mm->add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", s3.lens()}}), l2);
+        auto l2_ct = mm->add_instruction(migraphx::make_op("contiguous"), l2_mb);
+        auto pw0   = add_pointwise(p1, "main:pointwise0", {l2_ct, x1_rsp}, single_pointwise("add"));
+        auto pw0_rsp =
+            mm->add_instruction(migraphx::make_op("reshape", {{"dims", s1.lens()}}), pw0);
+        auto pw1   = add_pointwise(p1, "main:pointwise1", {x2, pw0_rsp}, single_pointwise("mul"));
+        auto pw2   = add_pointwise(p1, "main:pointwise2", {pw1}, single_pointwise("log"));
+        auto rsum0 = mm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", {2}}}), pw2);
+        auto rsum0_mb = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), rsum0);
+        auto rsum0_ct = mm->add_instruction(migraphx::make_op("contiguous"), rsum0_mb);
+        auto pw3 =
+            add_pointwise(p1, "main:pointwise3", {pw0_rsp, rsum0_ct}, single_pointwise("add"));
+        auto pw4    = add_pointwise(p1, "main:pointwise4", {pw3}, single_pointwise("log"));
+        auto rsum1  = mm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", {2}}}), pw4);
+        auto pw5    = add_pointwise(p1, "main:pointwise5", {rsum1}, single_pointwise("log"));
+        auto pw5_mb = mm->add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), pw5);
+        auto pw5_ct = mm->add_instruction(migraphx::make_op("contiguous"), pw5_mb);
+
+        auto l1_mb =
+            mm->add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), l1);
+        auto pw6 = add_pointwise(p1, "main:pointwise6", {pw5_ct, l1_mb}, single_pointwise("mul"));
+
+        auto rsum2 = mm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", {1}}}), pw6);
+        mm->add_return({rsum2});
+    }
+    run_pass(p1);
+    migraphx::program p2;
+    {
+        auto* mm = p2.get_main_module();
+        auto x1  = mm->add_parameter("x1", s1);
+        auto x2  = mm->add_parameter("x2", s1);
+        auto l1  = mm->add_literal(migraphx::literal(s2, {1.0}));
+        auto l2  = mm->add_literal(migraphx::literal(s2, {2.0}));
+
+        auto l1_mb =
+            mm->add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), l1);
+        auto l2_mb =
+            mm->add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), l2);
+
+        auto reduce0 = add_reduce(
+            p2,
+            "main:pointwise0:main:pointwise1:main:reduce_sum1:main:pointwise2:main:reduce_sum0:"
+            "main:pointwise3:main:pointwise4:main:pointwise5:main:pointwise6_reshape_reshape",
+            {l2_mb, x1, x2, l1_mb},
+            {2},
+            [&](auto* rm, const auto& inputs, const auto& axes) {
+                auto pw0 = add_pointwise(
+                    p2, rm, "main:pointwise0", {inputs[0], inputs[1]}, single_pointwise("add"));
+                auto pw1 = add_pointwise(
+                    p2, rm, "main:pointwise1", {inputs[2], pw0}, single_pointwise("mul"));
+                auto pw2 = add_pointwise(p2, rm, "main:pointwise2", {pw1}, single_pointwise("log"));
+                auto rsum0 =
+                    rm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", axes}}), pw2);
+                auto rsum0_mb = rm->add_instruction(
+                    migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), rsum0);
+                auto pw3 = add_pointwise(
+                    p2, rm, "main:pointwise3", {pw0, rsum0_mb}, single_pointwise("add"));
+                auto pw4 = add_pointwise(p2, rm, "main:pointwise4", {pw3}, single_pointwise("log"));
+                auto rsum1 =
+                    rm->add_instruction(migraphx::make_op("reduce_sum", {{"axes", axes}}), pw4);
+                auto pw5 =
+                    add_pointwise(p2, rm, "main:pointwise5", {rsum1}, single_pointwise("log"));
+                auto pw5_mb = rm->add_instruction(
+                    migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), pw5);
+                return add_pointwise(
+                           p2, rm, "main:pointwise6", {pw5_mb, inputs[3]}, single_pointwise("mul"));
+            });
+
+        auto reduce1 =
+            add_reduce(p2,
+                       "main:reduce_sum2_reshape",
+                       {reduce0},
+                       {1},
+                       [&](auto* rm, const auto& inputs, const auto& axes) {
+                           return rm->add_instruction(
+                               migraphx::make_op("reduce_sum", {{"axes", axes}}), inputs[0]);
+                       });
+
+        mm->add_return({reduce1});
+    }
+
     EXPECT(p1.sort() == p2.sort());
 }
 

--- a/test/fuse_reduce.cpp
+++ b/test/fuse_reduce.cpp
@@ -1014,7 +1014,7 @@ TEST_CASE(reshape_reduce_reduce_reduce_diff_axes)
                 auto pw5_mb = rm->add_instruction(
                     migraphx::make_op("multibroadcast", {{"out_lens", s1.lens()}}), pw5);
                 return add_pointwise(
-                           p2, rm, "main:pointwise6", {pw5_mb, inputs[3]}, single_pointwise("mul"));
+                    p2, rm, "main:pointwise6", {pw5_mb, inputs[3]}, single_pointwise("mul"));
             });
 
         auto reduce1 =

--- a/test/fuse_reduce.cpp
+++ b/test/fuse_reduce.cpp
@@ -895,7 +895,7 @@ TEST_CASE(reduce_reshape_reduce)
         auto y   = mm->add_parameter("y", s2);
         auto x1r = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), x1);
         auto x2r = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3r.lens()}}), x2);
-        auto yr  = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), y);
+        auto yr      = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), y);
         auto freduce = add_reduce(
             p2,
             "main:pointwise2:main:reduce_sum2_reshape_reshape:main:pointwise3_reshape:main:reduce_"

